### ISSUE TITLE
`no-conflicting-classes` false positive with `text-` and `tracking-`

### ISSUE THREAD

**Issue by tleunen** [sergioazoc/oxlint-tailwindcss#8]
```
"text-base" and "tracking-tight" affect "letter-spacing". "tracking-tight" takes precedence (appears later).oxc(tailwindcss(no-conflicting-classes))
```

text-base and tracking-tight are defined like this, so it's ok to have both at the same time, `tracking-tight` is done to override the letter-spacing only

```
.text-base {
  font-size: var(--text-base) /* 16px */;
  line-height: var(--tw-leading, var(--text-base--line-height) /* 24px */);
  letter-spacing: var(--tw-tracking, var(--text-base--letter-spacing) /* 0 */);
}

.tracking-tight {
  --tw-tracking: var(--tracking-tight) /* -0.025em */;
  letter-spacing: var(--tracking-tight) /* -0.025em */;
}
```

**Comment by sergioazoc** [comment]
Fixed in v0.4.1.

This happens when the theme defines `--text-base--letter-spacing` (or similar), which makes `text-base` generate `letter-spacing` in its CSS output. The rule already had a composition pair for `[text-*, leading-*]` (line-height override) but was missing the analogous pair for `[text-*, tracking-*]` (letter-spacing override).

The fix adds `[/^text-/, /^tracking-/]` to the `COMPOSITION_PAIRS` array. With the default Tailwind theme (no custom letter-spacing), `text-base` doesn't generate `letter-spacing` at all, which is why this only manifested with custom themes.

Thanks for the report!
